doc/Makefile

```diff
@@ -16,6 +16,16 @@ DT_TURBO_MODE ?= 0
 html-fast:
 	${MAKE} html DT_TURBO_MODE=1
 
+# The `minimal` tag turns off all slow sphinx features for a much faster
+# incremental build when editing .rst files interactively. As usual when
+# making significant sphinx configuration changes, you must delete
+# doc/_build/ first. WARNING: doxygen integration (breathe) will be
+# missing, so building from scratch will print a LOT of warnings!
+# However this is very useful for small, drive-by .rst fixes.
```

Comment:
I don't think a build with _lots_ of warnings is a good idea. All _fast_ modes we've had until now have produced clean builds, so I think we should do the same here.

Reply:
> All fast modes we've had

Any other mode besides `html-fast`? Which is... too slow for interactive use.

This new mode does not have to be official or documented. I can remove the new target and go back to just the sphinx tag.

What alternative(s) do you recommend?

Reply:
> > All fast modes we've had
> 
> Any other mode besides `html-fast`? Which is... too slow for interactive use.
> 
> This new "minimal" mode does not have to be official or documented. I can remove the new target and go back to just the sphinx tag. We just need something usable that does not require a code change.
> 

Why not official or documented?

> What alternative(s) do you recommend?

Solve the warnings problem, or, try to contribute upstream to improve the extensions that are causing this issue. I also think that if an extension requires us to do such hackery, we should probably reconsider its usage unless it can be fixed.

Reply:
> Why not official or documented?

Because disabling breathe triggers a lot of breathe warnings (which are harmless when you don't care about breathe and just want to make some .rst changes)

>  try to contribute upstream to improve the extensions that are causing this issue. I also think that if an extension requires us to do such hackery, we should probably reconsider its usage unless it can be fixed.

This PR provides a massive improvement with just one `if` conditional. It's well tested and working now. These alternatives represent at least months of work for uncertain results. They're not really "alternatives".

> ... or solve the warnings problem

I could try to filter them out, would that work for you?

Reply:
I'd be okay with filtering the doxygen and kconfig warnings.  The documentation should clearly state that the "minimal" option bypasses all doxygen and kconfig reference generation.

Reply:
> I'd be okay with filtering the doxygen and kconfig warnings. 

Thanks, I will try to find some time and do that in July.